### ISSUE TITLE
Problem with nonlocal written only falsely reported as unused.

### ISSUE THREAD

**Issue by kayhayen** [PyCQA/pylint#3044]
### Steps to reproduce

The above code in a Nuitka test gives a warning:

```python
def simpleFunction4():
    a = 1

    def nonlocal_writer():
        nonlocal a

        for a in range(10): 
            pass

    nonlocal_writer()

    assert a == 9, a
```

### Current behavior

I need to put # false alarm, pylint: disable=unused-variable in the loop.

### Expected behavior

Since the nonlocal value is being read, it shouldn't be considered unused.

### pylint --version output

$ /c/Python37/python -m pylint --version
__main__.py 2.3.1
astroid 2.2.5
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 22:22:05) [MSC v.1916 64 bit (AMD64)]

**Comment by PCManticore** [comment]
Thanks @kayhayen for the report. This looks like a slight edge case though as you're relying on the side effect of iteration to set the `a` variable in the outer scope. Still a bug though.

**Comment by kayhayen** [comment]
It's from a Nuitka test, I don't think it deserves very much priority, but I think there is something flawed then, a potential write, still is a write, right?
